### PR TITLE
Fix the issue of blip model returning loss even when the label is not provided. 

### LINKED ISSUE #21510: BLIP computes loss no matter what
### System Info

transformers-4.27.0.dev0

### Who can help?

@younesbelkada 

### Reproduction

Currently, `BlipForConditionalGeneration` will always return a loss, even when the user hasn't provided any labels:

```
from PIL import Image
import requests
from transformers import AutoProcessor, BlipForConditionalGeneration

processor = AutoProcessor.from_pretrained("Salesforce/blip-image-captioning-base")
model = BlipForConditionalGeneration.from_pretrained("Salesforce/blip-image-captioning-base")

url = "http://images.cocodataset.org/val2017/000000039769.jpg"
image = Image.open(requests.get(url, stream=True).raw)

inputs = processor(images=image, return_tensors="pt")

outputs = model(**inputs)
print(outputs.keys())
```
which prints:
```
odict_keys(['loss', 'decoder_logits', 'image_embeds', 'last_hidden_state'])
```
And what's also weird is that this loss is NaN:
```
print(outputs.loss)
```
returns
```
tensor(nan, grad_fn=<AddBackward0>)
```

### Expected behavior

Returning a loss in case the user doesn't provide `labels` isn't really compliant to the design of all models in the Transformers library, which only return a loss in case the user provides labels.

Hence I'm advocating to remove [the following lines of code](https://github.com/huggingface/transformers/blob/5b67ab9924cf7587b39b59eb0bf0abd3d099e8b9/src/transformers/models/blip/modeling_blip.py#L1008-L1009). Similar to models like `T5ForConditionalGeneration`, `BartForConditionalGeneration`, it is the user's responsibility to create the appropriate labels.

Moreover, I'm not sure why [the following lines of code](https://github.com/huggingface/transformers/blob/5b67ab9924cf7587b39b59eb0bf0abd3d099e8b9/src/transformers/models/blip/modeling_blip.py#L1005-L1006) are included, as they don't seem necessary and also lead to unexpected behaviour. Again here, models like T5 and BART don't set the `input_ids` automatically to the decoder start token ID in case the user doesn't provide `input_ids`. It's the user's responsibility to provide `input_ids`. Instead, those lines should also simply be removed IMO.

### PR DESCRIPTION
**Pull request by raghavanone**
# What does this PR do?

Fixes #21510 

@NielsRogge

### CODE REVIEW

**Comment by sgugger** [comment]
cc @younesbelkada and @amyeroberts

File: src/transformers/models/blip/modeling_blip.py [hunk 1]
```diff
@@ -990,7 +990,7 @@ def forward(
 
         >>> outputs = model(**inputs)
         ```"""
-        batch_size = pixel_values.shape[0]
+        pixel_values.shape[0]
```
Comment (NielsRogge) [inline src/transformers/models/blip/modeling_blip.py:993]: this line can be removed as well

**Review by younesbelkada — COMMENTED** [review]
Thanks for the PR! This touches some core components of the forward pass, can you confirm us the slow tests pass? `RUN_SLOW=1 pytest tests/models/blip/test_modeling_blip.py`?

**Comment by raghavanone** [comment]
> 

@younesbelkada 
One of the test (test_inference_image_captioning_fp16) is failing, not sure if it is related to my changes . 

    def _conv_forward(self, input: Tensor, weight: Tensor, bias: Optional[Tensor]):
        if self.padding_mode != 'zeros':
            return F.conv2d(F.pad(input, self._reversed_padding_repeated_twice, mode=self.padding_mode),
                            weight, bias, self.stride,
                            _pair(0), self.dilation, self.groups)
>       return F.conv2d(input, weight, bias, self.stride,
                        self.padding, self.dilation, self.groups)
E       RuntimeError: "slow_conv2d_cpu" not implemented for 'Half'

**Comment by younesbelkada** [comment]
Hi @raghavanone 
Hum I just went through the daily CI report and this test seems to be not failing on our end, can you share with us the traceback of the error?

**Comment by raghavanone** [comment]
> Hi @raghavanone Hum I just went through the daily CI report and this test seems to be not failing on our end, can you share with us the traceback of the error?

tests/models/blip/test_modeling_blip.py:1113 (BlipModelIntegrationTest.test_inference_image_captioning_fp16)
self = <tests.models.blip.test_modeling_blip.BlipModelIntegrationTest testMethod=test_inference_image_captioning_fp16>

    def test_inference_image_captioning_fp16(self):
        model = BlipForConditionalGeneration.from_pretrained(
            "Salesforce/blip-image-captioning-base", torch_dtype=torch.float16
        ).to(torch_device)
        processor = BlipProcessor.from_pretrained("Salesforce/blip-image-captioning-base")
        image = prepare_img()
    
        # image only
        inputs = processor(images=image, return_tensors="pt").to(torch_device, torch.float16)
    
>       predictions = model.generate(**inputs)

tests/models/blip/test_modeling_blip.py:1124: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/opt/homebrew/Caskroom/miniforge/base/envs/hf_dev/lib/python3.8/site-packages/torch/autograd/grad_mode.py:27: in decorate_context
    return func(*args, **kwargs)
src/transformers/models/blip/modeling_blip.py:1068: in generate
    vision_outputs = self.vision_model(
/opt/homebrew/Caskroom/miniforge/base/envs/hf_dev/lib/python3.8/site-packages/torch/nn/modules/module.py:1194: in _call_impl
    return forward_call(*input, **kwargs)
src/transformers/models/blip/modeling_blip.py:694: in forward
    hidden_states = self.embeddings(pixel_values)
/opt/homebrew/Caskroom/miniforge/base/envs/hf_dev/lib/python3.8/site-packages/torch/nn/modules/module.py:1194: in _call_impl
    return forward_call(*input, **kwargs)
src/transformers/models/blip/modeling_blip.py:241: in forward
    patch_embeds = self.patch_embedding(pixel_values)  # shape = [*, width, grid, grid]
/opt/homebrew/Caskroom/miniforge/base/envs/hf_dev/lib/python3.8/site-packages/torch/nn/modules/module.py:1194: in _call_impl
    return forward_call(*input, **kwargs)
/opt/homebrew/Caskroom/miniforge/base/envs/hf_dev/lib/python3.8/site-packages/torch/nn/modules/conv.py:463: in forward
    return self._conv_forward(input, self.weight, self.bias)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = Conv2d(3, 768, kernel_size=(16, 16), stride=(16, 16))
input = tensor([[[[ 0.8647,  0.9229,  0.9375,  ...,  1.7549,  1.7549,  1.7549],
          [ 0.9082,  0.9375,  0.9521,  ...,  1....2856, -0.3569],
          [-0.3142, -0.3425, -0.3569,  ..., -0.3000, -0.3569, -0.3994]]]],
       dtype=torch.float16)
weight = Parameter containing:
tensor([[[[ 3.3875e-03,  1.4102e-04,  7.0906e-04,  ..., -4.6539e-03,
            1.2560e-03, -5....3.7727e-03,  ..., -1.3084e-03,
            4.8304e-04,  7.3357e-03]]]], dtype=torch.float16,
       requires_grad=True)
bias = Parameter containing:
tensor([ 7.6477e-02,  7.6233e-02,  2.6343e-01,  2.6718e-02,  3.8727e-02,
         1.2962e-02, -2....6932e-02, -9.2529e-02,
         7.5012e-02,  6.4812e-03, -1.7303e-02], dtype=torch.float16,
       requires_grad=True)

    def _conv_forward(self, input: Tensor, weight: Tensor, bias: Optional[Tensor]):
        if self.padding_mode != 'zeros':
            return F.conv2d(F.pad(input, self._reversed_padding_repeated_twice, mode=self.padding_mode),
                            weight, bias, self.stride,
                            _pair(0), self.dilation, self.groups)
>       return F.conv2d(input, weight, bias, self.stride,
                        self.padding, self.dilation, self.groups)
E       RuntimeError: "slow_conv2d_cpu" not implemented for 'Half'

/opt/homebrew/Caskroom/miniforge/base/envs/hf_dev/lib/python3.8/site-packages/torch/nn/modules/conv.py:459: RuntimeError

**Comment by younesbelkada** [comment]
I see, you are not running the tests on GPU, if you don't have access to any GPU I can run the slow test for you
(Also we might need to add `require_torch_gpu` decorator on this test, if you could also add it in this PR it would be great 🙏 )

**Comment by raghavanone** [comment]
> test_inference_image_captioning_fp16

Oh, I just realised that, I am adding the tag.

File: tests/models/blip/test_modeling_blip.py [hunk 2]
```diff
@@ -1111,6 +1111,7 @@ def test_inference_image_captioning(self):
             [30522, 1037, 3861, 1997, 1037, 2450, 3564, 2006, 1996, 3509, 2007, 2014, 3899, 102],
         )
 
+    @require_torch
```
Comment (younesbelkada) [inline tests/models/blip/test_modeling_blip.py:1114]:
```suggestion
    @require_torch_gpu
```

Comment (younesbelkada) [reply tests/models/blip/test_modeling_blip.py:1114]: you might need to import it above from `testing_utils`

Comment (raghavanone) [reply tests/models/blip/test_modeling_blip.py:1114]: Done

**Comment by raghavanone** [comment]
@younesbelkada  Did that test fail in your setup ? Is there something I have to fix ?

**Review by younesbelkada — APPROVED** [review]
Thanks a lot for fixing and for taking care of this issue 💯 ! 
I can confirm all slow tests pass with these changes
I am ok with this PR in the principe that the slow tests are passing + there should be strong tests to test that this does not break any changes we made w.r.t Blip fine-tuning (also tested this branch on the fine-tuning Gcolab)

**Review by amyeroberts — APPROVED** [review]
Thanks for the fix!

**Review by sgugger — APPROVED** [review]

**Pull request merged**